### PR TITLE
leofs: Switch logger to lager backend

### PR DESCRIPTION
**Pull request by windkit**
### Related PR
https://github.com/leo-project/leo_logger/pull/5
https://github.com/leo-project/lager/pull/1

### Related Issue
#729

### CODE REVIEW

File: apps/leo_gateway/rebar.config [hunk 1]
```diff
@@ -45,7 +45,9 @@
             warn_shadow_vars,
             warn_export_vars,
             warn_export_all,
-            {platform_define, "^[0-9]+", namespaced_types}
+%            {platform_define, "^[0-9]+", namespaced_types}
```
Comment (mocchira) [inline apps/leo_gateway/rebar.config]: remove the comment above

File: apps/leo_manager/rebar.config [hunk 2]
```diff
@@ -41,7 +41,9 @@
             warn_shadow_vars,
             warn_export_vars,
             warn_export_all,
-            {platform_define, "^[0-9]+", namespaced_types}
+%            {platform_define, "^[0-9]+", namespaced_types}
```
Comment (mocchira) [inline apps/leo_manager/rebar.config]: remove the comment above

File: apps/leo_storage/rebar.config [hunk 3]
```diff
@@ -41,7 +41,9 @@
             warn_shadow_vars,
             warn_export_vars,
             warn_export_all,
-            {platform_define, "^[0-9]+", namespaced_types}
+%            {platform_define, "^[0-9]+", namespaced_types}
```
Comment (mocchira) [inline apps/leo_storage/rebar.config]: remove the comment above

File: rel/common/launch.environment [hunk 4]
```diff
@@ -3,9 +3,12 @@
...
+
 # Directory for .schema file.
 # RUNNER_SCHEMA_DIR=
+
 # Directory for erlang log files (erlang.log.* and run_erl.log)
 # RUNNER_LOG_DIR=
+
 # Defaults to "leofs"
-# RUNNER_USER=
+RUNNER_USER=wilson
```
Comment (mocchira) [inline rel/common/launch.environment]: rollback this file

**Comment by mocchira** [comment]
also fix the commit comments by adding the prefix "leofs: " as both commits affect the whole components.

**Comment by windkit** [comment]
Access Log Benchmark with Lager
https://github.com/leo-project/notes/tree/master/leofs/benchmark/leofs/1.3/1.3.8/front/20171013_image_f4m_load_read_lager_leologger_ssd_30min_1

compare with LeoLogger without access log, performance is about the same

**Comment by mocchira** [comment]
@windkit the result LGTM. is there any update on leo_logger that you still don't push? if nothing, I will merege leo_logger then this one.

**Comment by windkit** [comment]
@mocchira I would have to test when one storage node is down too, please wait until that

**Comment by mocchira** [comment]
@windkit OK. let me know when you finished.

**Comment by windkit** [comment]
@mocchira The PRs should be ready to merge

**Comment by mocchira** [comment]
@windkit Change each rebar.config to refer to develop on leo_logger and run travis through github. Once all tests completed, I'll merge.

**Comment by windkit** [comment]
@mocchira Sorry that I forgot to check with the unit tests first and I have added a cleanup method to `lager_logger_api`
https://github.com/leo-project/leo_logger/pull/6

**Comment by mocchira** [comment]
@windkit please resolve conflicts happened on apps/leo_(gateway|manager|storage)/rebar.config

**Comment by yosukehara** [comment]
@windkit I've found `undefined error` from the CI's log as below:

```erlang
Setup Lager Logger API
undefined
*** context setup failed ***
**in function ets:new/2
  called as new(lager_logid_sink,[named_table,bag,public,{read_concurrency,true}])
in call from lager_logger_api:new/3 (src/lager_logger_api.erl, line 109)
in call from leo_storage_handler_object_tests:setup/0 (test/leo_storage_handler_object_tests.erl, line 82)
**error:badarg
undefined
*** context setup failed ***
**in function leo_storage_handler_object_tests:setup/0 (test/leo_storage_handler_object_tests.erl, line 77)
**error:{badmatch,{error,{already_running,'node_1@testing-gce-52fcb1d5-43bc-41b6-98c0-63fa9e814f60'}}}
```

**Comment by windkit** [comment]
@yosukehara it is because the setup is called multiple times but teardown is only called after all the tests are finished. `ets:new/2` in `lager_logger_api:new` would fail with `{error, badarg}`

**Comment by windkit** [comment]
It seems the setup is called multiple times before a teardown is called, I will check what happened

**Comment by windkit** [comment]
@yosukehara node stop is having problem, I am checking the issue

**Comment by windkit** [comment]
The node stop issue is fixed with https://github.com/leo-project/leo_logger/pull/7

Sorry that I forgot about the deadlock issue which I encountered previously

**Comment by mocchira** [comment]
@windkit Can you change the version of leo_logger to develop to pull the fix and run travis again?

**Comment by windkit** [comment]
@mocchira done

**Pull request merged**